### PR TITLE
[12.0][ADD] base_user_locale: configure locale settings

### PR DESCRIPTION
**Pull request by alexey-pelykh**
This module allows overriding following settings for user or company:
 * Time format
 * Date format
 * First day of the week

### CODE REVIEW

**Review by moylop260 — APPROVED** [review]
LGTM
Thanks

**Comment by alexey-pelykh** [comment]
ping @andriisem @naglis @kamleshahir @xabispacebiker just in case you need it

**Comment by alexey-pelykh** [comment]
@lreficent @tarteo I'd like to kindly request your reviews, if possible

**Review by tarteo — APPROVED** [review]
LGTM. Very nice feature!

Can you increase coverage?

**Comment by alexey-pelykh** [comment]
@tarteo if you'll give a hint how to test-cover WebController route :)

File: base_user_locale/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (hieulucky111) [inline base_user_locale/models/res_company.py]: order  import, should be `fields, models`.

File: base_user_locale/models/res_users.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (hieulucky111) [inline base_user_locale/models/res_users.py]: order import

**Review by hieulucky111 — APPROVED** [review]
LGTM overall, i just asked some minor changed.

File: base_user_locale/models/res_company.py (hunk 1 above)
Comment (alexey-pelykh) [reply base_user_locale/models/res_company.py]: done

File: base_user_locale/models/res_users.py (hunk 2 above)
Comment (alexey-pelykh) [reply base_user_locale/models/res_users.py]: done

**Comment by alexey-pelykh** [comment]
@moylop260 can be merged, right?

**Review by artemsenko — APPROVED** [review]

**Comment by tarteo** [comment]
> @tarteo if you'll give a hint how to test-cover WebController route :)

Like this:
https://github.com/OCA/website/blob/11.0/website_adv_image_optimization/tests/test_controller.py

**Comment by alexey-pelykh** [comment]
@tarteo unfortunately, super() call will fail, there's no valid session during test :(

**Comment by moylop260** [comment]
@alexey-pelykh @tarteo 
Should we merge after your last comment?

**Comment by moylop260** [comment]
@rousseldenis  

How we can cancel the merge command?

**Comment by alexey-pelykh** [comment]
@moylop260 it won't merge if tests fail, so no worries

**Comment by alexey-pelykh** [comment]
@moylop260 see, all good :) now we can merge!

**Comment by alexey-pelykh** [comment]
@tarteo I've did what I could to enhance code coverage

**Comment by tarteo** [comment]
@alexey-pelykh Thanks!

**Pull request merged**